tests/other/test_pthread_icu.cpp

```diff
@@ -0,0 +1,13 @@
+#include <thread>
+#include <iostream>
+#include <unicode/unistr.h>
+
+// Test that code using both pthread and icu compiles.
+int main () {
+	std::thread([] {
+		std::string str;
+		icu::UnicodeString ustr("Hello world!");
+		ustr.toUTF8String(str);
+		std::cout << str << std::endl;
+	}).detach();
```

Comment:
Better to `.join` here since emscripten doesn't handling the case where the main application thread exits while other threads are still running.

Probably best to use `PROXY_TO_PTHREAD` and drop `PTHREAD_POOL_SIZE_STRICT`